File: Java LAB/Semester_2/src/Queue.java
import java.util.*;
public class Queue {
       public static void main(String[] args)
       {
           Scanner scan  = new Scanner(System.in);
           int size,front=-1,rear=-1;
           System.out.println("Eneter the size of the Queue");
           size = scan.nextInt();
           int[] queue = new int[size];
           int choice;
           boolean flag=true;
           while(flag)
           {
               System.out.println("Enter your choice to perform following actions");
               System.out.println("1. Insert the element into the queue");
               System.out.println("2. Delete the element from the queue");
               System.out.println("3. Display all the elements of the queue");
               System.out.println("4. Exit the program");
               System.out.println("5. Value of rear and front");
               choice = scan.nextInt();
               switch(choice)
               {
                   case 1:
                       if(rear == size-1)
                       {
                           System.out.println("Queue is full cannot insert element");
                       }
                       else
                       {
                           System.out.println("Enter element to insert into the queue");
                           queue[++rear]=scan.nextInt();
                           if(rear==0)
                               front++;
                       }
                       break;
                   case 2:
                       if(front==-1)
                           System.out.println("Queue is empty");
                       else
                       {
                            System.out.println("Deleted element is = " + queue[front++]);
                            if(front>rear)
                                front=rear=-1;
                       }
                       break;
                   case 3:
                       if(front==-1)
                            System.out.println("Queue is empty");
                       else
                       {
                           System.out.println("Elements in the queue are");
                           for(int i=front;i<=rear;i++)
                               System.out.println(queue[i]);
                           System.out.println();
                       }
                       break;
                   case 4:
                       flag=false;
                       break;
                   case 5:
                       System.out.println("Front = " + front + " Rear = " + rear);
                       break;
                   default:
                       System.out.println("Incorrect choice please try again");
               }        
           }
       }
}

File: Data Structures/Queue.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{
	int front=-1,rear=-1,size;
	cout<<"Enter the no. of elements in the Queue"<<endl;
	cin>>size;
	int ar[size];
	while(1)
	{
		int choice;
		cout<<endl<<endl<<"---------------------"<<endl;
		cout<<"1. Insert element in the Queue"<<endl;
		cout<<"2. Delete element from the Queue"<<endl;
		cout<<"3. Display all elements in the Queue"<<endl;
		cout<<"4. Exit Program"<<endl<<endl;
		cin>>choice;
		switch(choice)
		{
			case 1:
				{
					if(front ==size-1 && rear ==size-1)
						cout<<endl<<"Element cannot be inserted 'Queue is full'"<<endl<<endl;
					else
					{
						int temp;
						cout<<"Enter element to be inserted"<<endl;					
						if(front ==-1 && rear == -1)
						{
							front++;
							rear++;
							cin>>ar[rear];
						}
						else
						{
							cin>>ar[++rear];
						}
					}
					break;		
				}
			case 2:
				{
					if(front ==-1 && rear ==-1)
						cout<<endl<<"No element to delete 'Queue is empty'"<<endl<<endl;
					else
					{
						cout<<"Deleted element is"<<ar[front++]<<endl<<endl;					
						if(rear == -1 || (front==rear+1))
						{
							front=-1;
							rear=-1;
						}
					}
					break;		
				}
			case 3:
				{
					if(front==-1 && rear==-1)
						cout<<endl<<endl<<"No elements to Display 'Queue is empty'"<<endl<<endl;
					else
					{
						int temp1=front, temp2=rear;
						cout<<endl<<endl<<"Elements in the Queue are "<<endl<<endl;
						while(temp1<=temp2)
						{
							cout<<ar[temp1++]<<"  ";
						}
						cout<<endl<<endl;	
					}
					break;
				}
			case 4:
				{
					exit(0);
				}
			default:
			{
				cout<<endl<<endl<<"Incorrect option please try again "<<endl<<endl;
			}	
		}
	}
}
